packages/molecule/src/cli.ts

```diff
@@ -24,21 +41,83 @@ const fileConfig: Partial<Config> = (() => {
 const config: Config = {
   objectKeyFormat: fileConfig.objectKeyFormat || "keep",
   prepend: fileConfig.prepend || "",
-  schemaFile: fileConfig.schemaFile || "schema.mol",
+  schemaFile: fileConfig.schemaFile,
+  schemaDir: fileConfig.schemaDir,
+  outDir: fileConfig.outDir,
 };
 
-// check if the schema file exists
-if (!fs.existsSync(config.schemaFile)) {
-  console.error(
-    `Schema file ${config.schemaFile} does not exist. Please configure the \`schemaFile\` in ${DEFAULT_CONFIG_FILE_NAME}`
-  );
-  process.exit(1);
-}
-
-const generated = codegen(fs.readFileSync(config.schemaFile, "utf-8"), {
+const codegenOption = {
   prepend: config.prepend,
   formatObjectKeys:
     config.objectKeyFormat === "camelcase" ? camelcase : undefined,
-});
+};
+
+if (config.schemaFile) {
+  outputSingleToConsole(config.schemaFile);
+} else if (config.outDir && config.schemaDir) {
+  buildProject(config.schemaDir, config.outDir);
+} else {
+  throw new Error(
+    `Invalid configuration. Please provide either schemaFile or schemaDir and outDir`
+  );
+}
+
+function buildProject(schemaDir: string, outDir: string) {
+  // https://stackoverflow.com/a/69867053
+  // $1 imported variables
+  // $2 quotes used for the import
+  // $3 import path
+
+  const importStatementRegex =
+    // eslint-disable-next-line no-useless-escape
+    /import([ \n\t]*(?:[^ \n\t\{\}]+[ \n\t]*,?)?(?:[ \n\t]*\{(?:[ \n\t]*[^ \n\t"'\{\}]+[ \n\t]*,?)+\})?[ \n\t]*)from[ \n\t]*(['"])([^'"\n]+)(?:['"])/;
 
-console.log(generated);
+  // get the parsed generated typescript path for a given mol file
+  function getOutputPath(molPath: string): string {
+    const outputRelativePath = molPath
+      .replace(schemaDir, "") // do not include the schemaDir in the output path
+      .replace(/.mol$/, ".ts"); // replace the file extension
+
+    return path.join(outDir, outputRelativePath);
+  }
+
+  // remove the trailing slash for both windows and *nix
+  schemaDir = schemaDir.replace(/[\\/]+$/, "");
+
+  const namedSchemas = globSync(
+    path.join(schemaDir, "**/*.mol")
+  ).map<ProjectSchemaOptions>((molPath) => {
+    const prepend = (() => {
+      const match = config.prepend.match(importStatementRegex);
```

Comment:
What is `prepend` used for in cli, will it allow multiple lines of import statements?

Reply:
Currently, it is not supported to have custom schema items in separate files. It is recommended that you put all custom schema items into one file for unified management. Here is an example of how you can do this:

`number.ts`

```ts
export declare const Uint32Codec: AnyCodec;  
```

`string.ts`

```ts
export declare const StringCodec: AnyCodec;
```

If you want to override `Uint32` and `String` in all generated targets, you can create a `primitive.ts` file to export them from one entry point:

```json
{
  "prepend": "export { Uint32, String } from './primitive';"
}
```

Then, in `primitive.ts`, you can export `Uint32` and `String` as follows:

```ts
export { Uint32Codec as Uint32 } from './number'
export { StringCodec as String } from './string'
```